cont3xt/vueapp/src/components/pages/Cont3xt.vue

```diff
@@ -742,6 +772,39 @@ export default {
           target.style = `margin-top: -${delta}px`;
         }
       });
+    },
+    setViewByQueryParamName (viewName) {
+      const removeViewParam = () => {
+        this.$router.push({ query: { ...this.$route.query, view: undefined } });
+      };
+
+      if (typeof viewName !== 'string' || viewName === '') {
+        console.log(`WARNING -- Invalid view '${viewName}' from query parameter... defaulting to current integrations.`);
+        removeViewParam();
+        return;
+      }
+      const lowercaseViewName = viewName.toLowerCase();
+      for (const view of this.getViews) {
+        if (view.name.toLowerCase() === lowercaseViewName) {
+          this.$store.commit('SET_SELECTED_VIEW', view.name);
+          this.$store.commit('SET_SELECTED_INTEGRATIONS', view.integrations);
+          return;
+        }
+      }
+      removeViewParam();
```

Comment:
why remove the view if it is a good view? If we left it, it would be easier to copy paste the url WITH view.

Reply:
My initial thought was to drop it after initial load of shared link click through, but now I think it *would* be better to retain it in the query string.

Reply:
The view query parameter is now only dropped when it is invalid or is not found among the users views (system defaults (all/none) are also safe).